src/components/NewCommentForm.tsx

```diff
@@ -59,13 +59,14 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
 
     setSubmitting(true);
 
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
+    try {
+      await onSubmit({ name, email, body });
+      clearForm();
+    } finally {
+      setSubmitting(false);
+    }
 
-    // and the spinner will disappear immediately
-    setSubmitting(false);
     setValues(current => ({ ...current, body: '' }));
```

Comment:
This line is redundant because `clearForm()` is already called, which resets the form values. Consider removing it to avoid unnecessary state updates.